### ISSUE TITLE
Inconsistent library filename?

### ISSUE THREAD

**Issue by rhattersley** [tinyendian/lfric_reader#4]
Having built an image from `Singularity-headless` the example script `src/python/paraview_script.py` fails to find the shared object file for the plugin:
```
$ singularity exec paraview_headless.sif pvpython src/python/paraview_script.py
(   1.136s) [main thread     ]  vtkPVPluginLoader.cxx:519    ERR| vtkPVPluginLoader (0x55c22149af90): libnetCDFLFRicReader.so: cannot open shared object file: No such file or directory

Traceback (most recent call last):
  File "src/python/paraview_script.py", line 7, in <module>
    pvs.LoadPlugin(pluginPath+'libnetCDFLFRicReader.so', remote=False)
  File "/usr/local/lib/python3.6/site-packages/paraview/simple.py", line 1880, in LoadPlugin
    servermanager.LoadPlugin(filename, remote)
  File "/usr/local/lib/python3.6/site-packages/paraview/servermanager.py", line 2381, in LoadPlugin
    raise RuntimeError ("Problem loading plugin %s" % (filename))
RuntimeError: Problem loading plugin libnetCDFLFRicReader.so
```

Checking the image contents shows a different filename:
```
$ singularity shell paraview_headless.sif.root
Singularity> ls /usr/local/lib/netCDFLFRicReader/
libvtkNetCDFLFRicReader.so  netCDFLFRicReader.so  vtk
```

Modifying the example script to use the filename in the image fixes the problem:
```python
pvs.LoadPlugin(pluginPath+'netCDFLFRicReader.so', remote=False)
```

Presumably the filename currently in the example script works on some systems when _not_ built within a Singularity container? It would be nice to have the example script work in both circumstances. I'm not familiar with ParaView builds, configuration, etc. but one simple option would be to define an environment variable in the Singularity image that is picked up by the example script when present.

So in the Singularity definition files we could add:
```
%environment
    ... <existing variable definitions >
    export PV_NC_PLUGIN_PATH=netCDFLFRicReader.so
```

And the start of the example script would become:
```python
import os

import paraview.simple as pvs


# Load the LFRic reader plugin - it may be necessary to provide the full path to
# libnetCDFLFRicReader.so, depending on the installation. Set remote=True if you
# connect ParaView with a separate server.
pluginPath = os.environ.get('PV_NC_PLUGIN_PATH', 'libnetCDFLFRicReader.so')
pvs.LoadPlugin(pluginPath, remote=False)
```

**Comment by rhattersley** [comment]
The lack of a `lib` prefix seems to have been spotted elsewhere - see "Scenario 3" in https://discourse.paraview.org/t/building-plugins-modules-with-paraview-5-7-0-cmake/3323

**Comment by rhattersley** [comment]
I'm guessing the difference in plugin filename is based on the version of ParaView, as that dictates [which build pattern to use](https://github.com/tinyendian/lfric_reader/blob/7432155dd8331d2f9189f5d6b57afe5067774c68/src/cxx/CMakeLists.txt#L15).

**Comment by tinyendian** [comment]
Hi @rhattersley many thanks for finding this issue, I had overlooked this example script when I updated the Python scripts in the `utils/paraviewmacros` repository on MOSRS to work with the Singularity containers and the latest ParaView release v5.8.

As you already said, the problem roots in a complete overhaul of the ParaView build system in v5.7, which resulted in two shared libraries being built for each plugin with a different naming convention compared to earlier releases, which used only a single `libnetCDFLFRicReader.so`.

The containerised ParaView builds are set up to automatically load the plugin, which is achieved by setting ParaView's environment variable `export PV_PLUGIN_PATH=/usr/local/lib/netCDFLFRicReader`, so it should not be necessary to use the `LoadPlugin` method there at all.

I understood that, at the Met Office, ParaView v5.5 is used, and the LFRic Reader plugin is installed in a way that does not require a full path to the shared library. So, for the MOSRS scripts, I used the following approach which attempts to load the plugin with pre-v5.7 naming convention only if it is not already loaded:
`
if not 'NetCDFLFRicReader' in dir(pvs):
    pvs.LoadPlugin('libnetCDFLFRicReader.so')
`
Even better would be to add a further test that loads the plugin using pre-v5.7 and v5.7-or-later naming convention.

If that works for you, I'll implement the required changes in the script.
